model/riscv_insts_zcmp.sail

```diff
@@ -0,0 +1,224 @@
+/*=======================================================================================*/
+/*  This Sail RISC-V architecture model, comprising all files and                        */
+/*  directories except where otherwise noted is subject the BSD                          */
+/*  two-clause license in the LICENSE file.                                              */
+/*                                                                                       */
+/*  SPDX-License-Identifier: BSD-2-Clause                                                */
+/*=======================================================================================*/
+
+function clause currentlyEnabled(Ext_Zcmp) = hartSupports(Ext_Zcmp) & currentlyEnabled(Ext_Zca) & not(currentlyEnabled(Ext_Zcd))
+
+mapping zcmp_assembly_rlist : bits(4) <-> string = {
+  0x4 <-> "ra",
+  0x5 <-> "ra" ^ sep() ^ "s0",
+  0x6 <-> "ra" ^ sep() ^ "s0-s1",
+  0x7 <-> "ra" ^ sep() ^ "s0-s2",
+  0x8 <-> "ra" ^ sep() ^ "s0-s3",
+  0x9 <-> "ra" ^ sep() ^ "s0-s4",
+  0xa <-> "ra" ^ sep() ^ "s0-s5",
+  0xb <-> "ra" ^ sep() ^ "s0-s6",
+  0xc <-> "ra" ^ sep() ^ "s0-s7",
+  0xd <-> "ra" ^ sep() ^ "s0-s8",
+  0xe <-> "ra" ^ sep() ^ "s0-s9",
+  0xf <-> "ra" ^ sep() ^ "s0-s11",
+  // Using X names instead of ABI names.
+  0x4 <-> "x1",
+  0x5 <-> "x1" ^ sep() ^ "x8",
+  0x6 <-> "x1" ^ sep() ^ "x8-x9",
+  0x7 <-> "x1" ^ sep() ^ "x8-x9" ^ sep() ^ "x18",
+  0x8 <-> "x1" ^ sep() ^ "x8-x9" ^ sep() ^ "x18-x19",
+  0x9 <-> "x1" ^ sep() ^ "x8-x9" ^ sep() ^ "x18-x20",
+  0xa <-> "x1" ^ sep() ^ "x8-x9" ^ sep() ^ "x18-x21",
+  0xb <-> "x1" ^ sep() ^ "x8-x9" ^ sep() ^ "x18-x22",
+  0xc <-> "x1" ^ sep() ^ "x8-x9" ^ sep() ^ "x18-x23",
+  0xd <-> "x1" ^ sep() ^ "x8-x9" ^ sep() ^ "x18-x24",
+  0xe <-> "x1" ^ sep() ^ "x8-x9" ^ sep() ^ "x18-x25",
+  0xf <-> "x1" ^ sep() ^ "x8-x9" ^ sep() ^ "x18-x27",
+}
+
+function get_stack_adj_base(rlist : bits(4)) -> int = {
+  if xlen == 32 then
+    match rlist[3 .. 2] {
+      0b01 => 16,
+      0b10 => 32,
+      0b11 => if rlist == 0b1111 then 64 else 48,
+      _    => 0
```

Comment:
These `=> 0` cases should be unreachable right? Can you change them to `internal_error(__FILE__, __LINE__, ...)`?